server/src/test/scala/za/co/absa/atum/server/api/database/runs/functions/GetPartitioningCheckpointsIntegrationTests.scala

```diff
@@ -0,0 +1,49 @@
+package za.co.absa.atum.server.api.database.runs.functions
+
+import doobie.util.Read
+import za.co.absa.atum.server.ConfigProviderTest
+import za.co.absa.atum.model.dto.{CheckpointQueryDTO, PartitionDTO, PartitioningDTO}
+import za.co.absa.atum.server.api.TestTransactorProvider
+import za.co.absa.atum.server.api.database.PostgresDatabaseProvider
+import za.co.absa.atum.server.model.CheckpointFromDB
+import zio.test.Assertion.failsWithA
+import zio.{Scope, ZIO}
+import zio.test._
+
+import doobie.postgres.implicits._
+import doobie.postgres.circe.jsonb.implicits.jsonbGet
+import za.co.absa.atum.server.api.database.DoobieImplicits.Sequence._
+
+object GetPartitioningCheckpointsIntegrationTests extends ConfigProviderTest {
+
+  override def spec: Spec[TestEnvironment with Scope, Any] = {
+
+    val partitioningDTO1: PartitioningDTO = Seq(
+      PartitionDTO("stringA", "stringA"),
+      PartitionDTO("stringB", "stringB")
+    )
+
+    suite("GetPartitioningCheckpointsIntegrationTests")(
+      test("Returns expected sequence of Checkpoints with existing partitioning") {
+        val partitioningQueryDTO: CheckpointQueryDTO = CheckpointQueryDTO(
+          partitioning = partitioningDTO1,
+          limit = Some(10),
+          checkpointName = Some("checkpointName")
+        )
+        // Read[CheckpointFromDB] implicit validation
+        Read[CheckpointFromDB]
```

Comment:
Is this needed here? Implicits are resolved on compile time. Your code wouldn't compile if the read instance couldn't be constructed.

Reply:
I left it there because of the conversation that you and I had that we should keep it there for future references.